### PR TITLE
Automate Git repo file structure changes during upgrade

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#577 

*Description of changes:*

Support:
- v0.5.0 -> v0.6.x
- v0.6.0 -> v0.6.1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/623#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/workflows/upgrade.go
Comment (danbudris) [inline pkg/workflows/upgrade.go:201]: consider if we want to return the new diagnostics task here, which executes the support bundle on the workload and mgmt clusters

File: pkg/addonmanager/addonclients/files.go
Comment (danbudris) [inline pkg/addonmanager/addonclients/files.go:69]:
```suggestion
	// in older version (<= 0.5.0), flux-system and eksa-system folders are under the same parent directory
```
🖊️

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/623#pullrequestreview-804157244" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/623#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/addonmanager/addonclients/files.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/files.go:69]:
```suggestion
	// in older version (<= 0.5.0), flux-system and eksa-system folders are under the same parent directory
```

File: pkg/addonmanager/addonclients/files.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/files.go:96]: Can we use the writer directly for this?

File: pkg/addonmanager/addonclients/files.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/files.go:100]: I think we should call this method using the old spec, so the contents of the kustomize file is the same
So probably have the two specs (old and new) being passed to `UpdateLegacyFileStructure`?

File: pkg/addonmanager/addonclients/files.go
Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/files.go:19]: not really, for workload cluster created w 0.6.0 we still need to update its clusterConfigPath

File: pkg/addonmanager/addonclients/files.go
Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/files.go:96]: why? we only get resourceSpec as []bytes

File: pkg/addonmanager/addonclients/fluxaddonclient.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/fluxaddonclient.go:277]: nit: I wonder if we should move the logic of `SetDefaultGitOps` here or somewhere else (no need to change now, just a thought)

File: pkg/addonmanager/addonclients/files.go
Comment (g-gaston) [inline pkg/addonmanager/addonclients/files.go:130]: I think it should be enough with the first `Unmarshal` :)

**Review by g-gaston — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/623#pullrequestreview-804157244" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/623#pullrequestreview-804295467" title="Approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/623#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,g-gaston,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/addonmanager/addonclients/fluxaddonclient.go
Comment (jiayiwang7) [reply pkg/addonmanager/addonclients/fluxaddonclient.go:277]: yea we can refactor later - for now i added in two validations so we avoid more  SetDefaultGitOps in flux methods

**Pull request merged**
